operator/v1alpha1/types_etcdbackup.go

```diff
@@ -0,0 +1,102 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:resource:scope=Cluster
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//
+// # EtcdBackup provides configuration options and status for a one-time backup attempt of the etcd cluster
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type EtcdBackup struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec EtcdBackupSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +kubebuilder:validation:Optional
+	// +optional
+	Status EtcdBackupStatus `json:"status"`
+}
+
+type EtcdBackupSpec struct {
+	// PVCName specifies the name of the PersistentVolumeClaim which binds a PersistentVolume where the
+	// etcd backup file would be saved
+	// +kubebuilder:validation:Required
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="pvcName is immutable once set"
+	PVCName string `json:"pvcName"`
+}
+
+const (
+	// Etcd backup is running
+	Running EtcdBackupState = "Running"
```

Comment:
consistent with pods, this is Phase and it is discouraged.  We found that the state machine ended up with too many tuples and conditions worked better.  conditions also work with `oc wait`.

Reply:
Going to remove the EtcdBackupState and add as conditions instead.

Reply:
Actually looking around I don't see many other APIs that have their conditions defined in their respective types.go.
E.g for the controlplanemachineset API, they're defined in the operator repo:
https://github.com/openshift/cluster-control-plane-machine-set-operator/blob/main/pkg/controllers/controlplanemachineset/consts.go

At the risk of over/under designing the conditions, I'll hold off from defining them here and we can add add them with the controllers in cluster-etcd-operator.
Once we have the controller implementation PR ready that should give us a better idea of whether we need something more or less than the default `Complete, Succeeded, Failed, Skipped` conditions.

@tjungblu I can see you already have some different conditions defined so let me know if that works better.

Reply:
feel free to define them properly, I mostly copied the conditions from the upgradebackup controller